csrc/ir/nodes.cpp

```diff
@@ -689,6 +689,57 @@ std::vector<PolymorphicValue> ArrayConstruct::evaluate(
 
 NVFUSER_DEFINE_CLONE_AND_CREATE(ArrayConstruct)
 
+ReverseArray::ReverseArray(IrBuilderPasskey passkey, Val* output, Val* input)
+    : Expr(passkey) {
+  TORCH_INTERNAL_ASSERT(
+      std::holds_alternative<ArrayOf>(input->dtype().type),
+      "Cannot reverse a non-array type.");
+  TORCH_INTERNAL_ASSERT(
+      std::holds_alternative<ArrayOf>(output->dtype().type),
+      "Cannot reverse a non-array type.");
+  auto input_array_type = std::get<ArrayOf>(input->dtype().type);
+  auto output_array_type = std::get<ArrayOf>(output->dtype().type);
+  TORCH_INTERNAL_ASSERT(
```

Comment:
Looks like `auto out = newScalar(array->dtype());` already ensures the same dtype for input and output. Or is this API can also be directly used without go through `IrBuilder::reverseArrayExpr(Val* array)`? If that is the case, maybe a test can be added.

Reply:
agree.